### PR TITLE
LUCENE-9688: Hunspell: consider prefix's continuation flags when applying suffix

### PR DESCRIPTION
**Pull request by donnerpeter**
# Description

aff:
SFX s Y 1
SFX s 0 s .

PFX p Y 1
PFX p 0 0/s .

dic:
calorie/p

"calories" should be accepted

# Solution

When applying suffix "s" (which expects "s" flag to be set), check not only the word itself (which only has "p") flag, but the continuation flags of the just applied prefix "p" which has "s" as its continuation.

# Tests

`TestDependencies` updated

# Checklist

Please review the following and check all that apply:

- [x] I have reviewed the guidelines for [How to Contribute](https://wiki.apache.org/solr/HowToContribute) and my code conforms to the standards described there to the best of my ability.
- [x] I have created a Jira issue and added the issue ID to my pull request title.
- [x] I have given Solr maintainers [access](https://help.github.com/en/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork) to contribute to my PR branch. (optional but recommended)
- [x] I have developed this patch against the `master` branch.
- [x] I have run `./gradlew check`.
- [x] I have added tests for my changes.
- [ ] I have added documentation for the [Ref Guide](https://github.com/apache/lucene-solr/tree/master/solr/solr-ref-guide) (for Solr changes only).

### CODE REVIEW

File: lucene/analysis/common/src/java/org/apache/lucene/analysis/hunspell/Stemmer.java [hunk 1]
```diff
@@ -588,6 +585,12 @@ private boolean checkCondition(
     return stems;
   }
 
+  private boolean isFlagAppendedByAffix(int affixId, char flag) {
+    if (affixId < 0) return false;
```
Comment (dweiss) [inline lucene/analysis/common/src/java/org/apache/lucene/analysis/hunspell/Stemmer.java:589]: Add curly braces. I know, it's verbose but some folks on this list even went as far as to require it (the formatter doesn't care).

File: lucene/analysis/common/src/java/org/apache/lucene/analysis/hunspell/Stemmer.java [hunk 2]
```diff
@@ -472,9 +472,9 @@ private boolean checkCondition(
    * @param strippedWord Word the affix has been removed and the strip added
    * @param length valid length of stripped word
    * @param affix HunspellAffix representing the affix rule itself
-   * @param prefixFlag when we already stripped a prefix, we cant simply recurse and check the
-   *     suffix, unless both are compatible so we must check dictionary form against both to add it
-   *     as a stem!
+   * @param prefixId when we already stripped a prefix, we cant simply recurse and check the suffix,
```
Comment (dweiss) [inline lucene/analysis/common/src/java/org/apache/lucene/analysis/hunspell/Stemmer.java:475]: cant -> can't (I realize it's not you, but maybe correct while we're here).

**Review by dweiss — APPROVED** [review]
Otherwise LGTM.

File: lucene/analysis/common/src/java/org/apache/lucene/analysis/hunspell/Stemmer.java (hunk 1 above)
Comment (donnerpeter) [reply lucene/analysis/common/src/java/org/apache/lucene/analysis/hunspell/Stemmer.java:589]: Curly braces and 2 more lines? Oh pity, I love conciseness so much :(

Comment (dweiss) [reply lucene/analysis/common/src/java/org/apache/lucene/analysis/hunspell/Stemmer.java:589]: The discussion of whether braces in single-line statements in if/while/for loops are needed or not
is like religion - you can't end it, once started... I'm just relaying what has been said 
on the mailing list after the formatter folded code like this:
```
if (foo)
  doSomething()
else
  doSomethingElse()
```

Privately, I don't have a strong opinion. Depends on the context where it's at, I guess.

Comment (donnerpeter) [reply lucene/analysis/common/src/java/org/apache/lucene/analysis/hunspell/Stemmer.java:589]: In that case I'd also put braces (although I've seen people who don't). But my case is "then-branch is a simple `return` on the same line". Are braces also required in this case? And separate lines, like this?
```
if (affixId < 0) {
  return;
}
```

Comment (dweiss) [reply lucene/analysis/common/src/java/org/apache/lucene/analysis/hunspell/Stemmer.java:589]: Required is not the right word... Perhaps... advised. :) You are entitled to your own opinion (at least as long as this passes precommit checks!) - like I said, I relayed what's been mentioned on the mailing list. Some people just put braces everywhere and they are used to this consistency, even if it adds verbosity.

Comment (donnerpeter) [reply lucene/analysis/common/src/java/org/apache/lucene/analysis/hunspell/Stemmer.java:589]: Would it be acceptable to reformat the code when such a demand appears, or after I'm finished with these improvements? These "some people" might never need to look into Hunspell code.

Comment (dweiss) [reply lucene/analysis/common/src/java/org/apache/lucene/analysis/hunspell/Stemmer.java:589]: I think consistency in those braces (over verbosity) is a good rule in general, but ok.

**Pull request merged**